src/functions/withPreventDoubleClick/withPreventDoubleClick.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import debounce from 'lodash.debounce';
+
+/*
+    Prevent onPress from triggering multiple times if pressed too quickly.
+    Extends the touchable components like TouchableHighlight, Button etc.
+    NOTE: Doesn't work with buttons that navigate. To prevent the issue
+    with buttons that navigate see functions/navigateWithDebounce.
+*/
+
+const withPreventDoubleClick = (WrappedComponent) => {
+
+    class PreventDoubleClick extends React.PureComponent {
+
```

Comment:
Mitä toi funktio käytännössä tekee?

Reply:
wrappaa jonkun componentin (pitäis toimii ainaki touchableHighlight ja button yms en oo testannu muilla, sit siin on debounce joka estää sillee että vain joka 1000 millisec voi pressaa

Reply:
navigoinneis toi ei toimi nii siks eriksee toi navigateWithDebounce